Makefile

```diff
@@ -1,7 +1,7 @@
 travis:
 	python setup.py test --coverage \
 		--coverage-package-name=indictrans
-	flake8 --max-complexity 10 indictrans
+	flake8 --max-complexity 22 indictrans
```

Comment:
Mostly it will be in conditionals getting too much nested. So try to refactor code into simple functions and try to avoid duplicate codes. This is general suggestion I've not checked the code under question.